src/components/RangeSlider/RangeSlider.js

```diff
@@ -0,0 +1,424 @@
+
+    return rect.left + global.window.pageXOffset;
+  }
+
+  getSliderLength() {
+    const slider = this.sliderRef;
+    if (!slider) {
+      return 0;
+    }
+
+    const coords = slider.getBoundingClientRect();
+    return coords.width;
+  }
+
+  handleGenerator = ({ index, ...restProps }) => {
+    if (restProps.value === null) {
+      return null;
+    }
+
+    return <Handle {...restProps} key={index} />;
+  };
+
+  moveTo(value) {
+    const { handle, bounds } = this.state;
+    const nextBounds = [...bounds];
+    nextBounds[handle] = value;
+
+    this.onChange({
+      handle,
+      bounds: nextBounds
+    });
+  }
+
+  calcValue(offset) {
+    const { min, max } = this.props;
+    const ratio = Math.abs(Math.max(offset, 0) / this.getSliderLength());
+    const value = ratio * (max - min) + min;
+    return value;
+  }
+
+  calcValueByPos(position) {
+    const { thershold } = this.props;
+    const { handle, bounds } = this.state;
+    const pixelOffset = position - this.getSliderStart();
+    const nextValue = utils.trimAlignValue(
+      this.calcValue(pixelOffset),
+      handle,
+      this.props,
+      bounds,
+      thershold
+    );
+    return nextValue;
+  }
+
+  addDocumentTouchEvents() {
+    global.window.addEventListener("touchmove", this.onTouchMove);
+    global.window.addEventListener("touchend", this.onEnd);
+  }
+
+  addDocumentMouseEvents() {
+    global.window.addEventListener("mousemove", this.onMouseMove);
+    global.window.addEventListener("mouseup", this.onEnd);
+  }
+
+  removeDocumentEvents() {
+    global.window.removeEventListener("touchmove", this.onTouchMove);
+    global.window.removeEventListener("touchend", this.onEnd);
+
+    global.window.removeEventListener("mousemove", this.onMouseMove);
+    global.window.removeEventListener("mouseup", this.onEnd);
+  }
+
+  calcOffset(value) {
+    const { min, max } = this.props;
+    const ratio = (value - min) / (max - min);
+    return ratio * 100;
+  }
+
+  saveHandle(index, handle) {
+    this.handlesRefs[index] = handle;
+  }
+
+  render() {
+    const { bounds } = this.state;
+    const { disabled, min, max, className } = this.props;
+
+    const offsets = bounds.map(v => this.calcOffset(v));
+
+    const handleClassName = "RangeSlider-handle";
+    const handles = bounds.map((v, i) =>
+      this.handleGenerator({
+        className: classNames(handleClassName, {
+          [`${handleClassName}-${i + 1}`]: true
+        }),
+        offset: offsets[i],
+        value: v,
+        index: i,
+        min,
+        max,
+        disabled,
+        ref: h => this.saveHandle(i, h)
+      })
+    );
+
+    const track = bounds.slice(0, -1).map((_, index) => {
+      const i = index + 1;
+      return (
+        <Track
+          className="RangeSlider-track"
+          offset={offsets[i - 1]}
+          length={offsets[i] - offsets[i - 1]}
+          key={i}
+          disabled={disabled}
+        />
+      );
+    });
```

Comment:
Can you please move these `maps` to methods? E.g. `this.renderHandles` and `this.renderTrack`